adapters/repos/db/lsmkv/store.go

```diff
@@ -41,6 +41,7 @@ type Store struct {
 	// Prevent concurrent manipulations to the bucketsByNameMap, most notably
 	// when initializing buckets in parallel
 	bucketAccessLock sync.RWMutex
+	bucketByNameLock map[string]*sync.Mutex
```

Comment:
**suggestion:** why not we just have single mutex to access the whole `bucketsByName` map ?  

also this pattern `*sync.Mutex` could cause us problems, because we may try to lock nil pointer  
```
map[string]*sync.Mutex
```

Reply:
Wouldn't that essentially reduce the maximum concurrency of store creations to 1?

Reply:
it will increase the contention for sure, we could use instead `sync.Map` or `map[string]sync.Mutex` without pointer, the idea here to avoid using pointer to mutex

Reply:
> it will increase the contention for sure, we could use instead `sync.Map` or `map[string]sync.Mutex` without pointer or a `channel` , the idea here to avoid using pointer to mutex

a thread-safe map is not enough in this case,  the slot is being locked while the bucket is instantiated. A lockableMap or some wrapping struct on the Bucket could be alternatives, but this was one of the simple implementations